### PR TITLE
Update the defn macro

### PR DESCRIPTION
**Pull request by yenda**
This macro should make the events.cljs ns obsolete over time.

the defn macro knows register the function as an events under the
keywords provided in the :events key of the attributes map. It also
adds the interceptors provided in the :interceptors map

exemple:

```clojure
(fx/defn hello4
  {:doc "this function is useless as well"
   :events [:test/valid1 :test/valid2]}
  [{:keys [db]} b]
  {:db (assoc db :a b) :b (:a db)})
```

## Testing

Tested in Android emulator.

This PR is more of a dev experience improvement, but since the macro is basically behind everything happening in status-react, it is still worth testing. 
In this PR it is used in the login events, so to test it:
- create an account, 
- logout play around a bit. 

I don't think the app would even be able to login if there was a bug in this PR

status: ready

### CODE REVIEW

File: test/cljs/status_im/test/utils/fx.cljs
Comment (dmitryn) [inline test/cljs/status_im/test/utils/fx.cljs:22]: @yenda  If i set multiple events like in this case, does it mean `hello4` would be called when `test/valid1`, `:test/valid2` gets dispatched? What is the use case for having multiple events handled by same defn?
Also, Is it possible to know what event triggered the call inside `hello4` body? Seems like event-name is not passed to the function.

Comment (yenda) [reply test/cljs/status_im/test/utils/fx.cljs:22]: sure I will add that. 

The reason for supporting multiple events is to encourage the use of names that describe what happens for events and not actions. So the function can be an action and multiple events can trigger it. For instance navigate-back can be triggered by wallet/back-button-pressed and profile/back-button-pressed

**Review by dmitryn — APPROVED** [review]

File: test/cljs/status_im/test/utils/fx.cljs
Comment (yenda) [reply test/cljs/status_im/test/utils/fx.cljs:22]: So I've looked into it and actually I think for the rare events that need to be self aware it is better to rely on an interceptor

**Review by flexsurfer — APPROVED** [review]

File: test/cljs/status_im/test/utils/fx.cljs
Comment (dmitryn) [reply test/cljs/status_im/test/utils/fx.cljs:22]: Ok, that makes sense not to add event-name to args as it's rarely used. It can be added to cofx by interceptor when needed as you said.

**Comment by yenda** [comment]
I added a test commit because this is hardcore macro chemistry and I wanted to make sure it worked so it is intended to be removed (or not) but will help with testing

File: src/status_im/utils/fx.clj
Comment (rasom) [inline src/status_im/utils/fx.clj]: it seems that we don't want to use `find-ns` anymore

**Review by rasom — CHANGES_REQUESTED** [review]

File: src/status_im/utils/fx.clj
Comment (yenda) [reply src/status_im/utils/fx.clj]: I don't think it can build without it before your PR is merged

Comment (rasom) [reply src/status_im/utils/fx.clj]: do you mean it will work only with `shadow-cljs`?

Comment (yenda) [reply src/status_im/utils/fx.clj]: I mean we will have warnings if I remove it now

**Comment by churik** [comment]
@rasom changes are still requested?

**Comment by churik** [comment]
Tested:
- creating account
- logout
- force logout actions (changing networks / bootnodes / log level)

File: src/status_im/utils/fx.clj
Comment (rasom) [reply src/status_im/utils/fx.clj]: Warning during development or in release builds?

Comment (rasom) [reply src/status_im/utils/fx.clj]: I don't see any warnings in dev mode

Comment (rasom) [reply src/status_im/utils/fx.clj]: I mean, without `find-ns`

Comment (yenda) [reply src/status_im/utils/fx.clj]: that is because `[status-im.utils.handlers :as handlers]` is already required in the only namespace (`status-im.accounts.login.core`) where the part that needs it is macro-expended.

Comment (yenda) [reply src/status_im/utils/fx.clj]: @rasom I removed it can you unblock the PR?

**Review by rasom — APPROVED** [review]

**Comment by yenda** [comment]
@churik the windows build failing isn't related, there is no more space on the node that built it @jakubgs

**Comment by yenda** [comment]
@churik is it ok to merge?

**Comment by churik** [comment]
@yenda yes, go ahead

**Pull request merged**
